src/python_testing/spec_parsing_support.py

```diff
@@ -362,96 +405,62 @@ def create_cluster(self) -> XmlCluster:
                           attributes=self.parse_attributes(),
                           accepted_commands=self.parse_commands(CommandType.ACCEPTED),
                           generated_commands=self.parse_commands(CommandType.GENERATED),
+                          unknown_commands=self.parse_unknown_commands(),
                           events=self.parse_events(), pics=self._pics)
 
     def get_problems(self) -> list[ProblemNotice]:
         return self._problems
 
 
-def build_xml_clusters() -> tuple[list[XmlCluster], list[ProblemNotice]]:
-    # workaround for aliased clusters not appearing in the xml. Remove this once https://github.com/csa-data-model/projects/issues/373 is addressed
-    conc_clusters = {0x040C: ('Carbon Monoxide Concentration Measurement', 'CMOCONC'),
-                     0x040D: ('Carbon Dioxide Concentration Measurement', 'CDOCONC'),
-                     0x0413: ('Nitrogen Dioxide Concentration Measurement', 'NDOCONC'),
-                     0x0415: ('Ozone Concentration Measurement', 'OZCONC'),
-                     0x042A: ('PM2.5 Concentration Measurement', 'PMICONC'),
-                     0x042B: ('Formaldehyde Concentration Measurement', 'FLDCONC'),
-                     0x042C: ('PM1 Concentration Measurement', 'PMHCONC'),
-                     0x042D: ('PM10 Concentration Measurement', 'PMKCONC'),
-                     0x042E: ('Total Volatile Organic Compounds Concentration Measurement', 'TVOCCONC'),
-                     0x042F: ('Radon Concentration Measurement', 'RNCONC')}
-    conc_base_name = 'Concentration Measurement Clusters'
-    resource_clusters = {0x0071: ('HEPA Filter Monitoring', 'HEPAFREMON'),
-                         0x0072: ('Activated Carbon Filter Monitoring', 'ACFREMON')}
-    resource_base_name = 'Resource Monitoring Clusters'
-    water_clusters = {0x0405: ('Relative Humidity Measurement', 'RH')}
-    water_base_name = 'Water Content Measurement Clusters'
-    aliases = {conc_base_name: conc_clusters, resource_base_name: resource_clusters, water_base_name: water_clusters}
-
-    def is_alias(id: uint):
-        for base, alias in aliases.items():
-            if id in alias:
-                return True
-        return False
+def add_cluster_data_from_xml(xml: ElementTree.Element, clusters: dict[int, XmlCluster], derived_clusters: dict[str, XmlCluster], ids_by_name: dict[str, int], problems: list[ProblemNotice]) -> None:
```

Comment:
Would be helpful if this was documented to know which are in and which are out parameters

Reply:
added desc.